java-api-core/src/com/kinvey/java/store/requests/data/CalculationRequest.java

```diff
@@ -0,0 +1,93 @@
+package com.kinvey.java.store.requests.data;
+
+import com.kinvey.java.KinveyException;
+import com.kinvey.java.Query;
+import com.kinvey.java.cache.ICache;
+import com.kinvey.java.model.AggregateEntity;
+import com.kinvey.java.model.Aggregation;
+import com.kinvey.java.network.NetworkManager;
+import com.kinvey.java.store.ReadPolicy;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+
+/**
+ * Created by yuliya on 10/06/17.
+ */
+
+public class CalculationRequest implements IRequest<Aggregation.Result[]> {
+    private final Query query;
+    private final AggregateEntity.AggregateType type;
+    private final ICache<Aggregation.Result> cache;
+    private ReadPolicy readPolicy;
+    private final NetworkManager<Aggregation.Result> networkManager;
+    private ArrayList<String> fields;
+    private final String field;
+
+    public CalculationRequest(AggregateEntity.AggregateType type, ICache cache, ReadPolicy readPolicy,
+                              NetworkManager networkManager,
+                              ArrayList<String> fields, String field, Query query) {
+        this.type = type;
+        this.cache = cache;
+        this.readPolicy = readPolicy;
+        this.networkManager = networkManager;
+        this.fields = fields;
+        this.field = field;
+        this.query = query;
+    }
+
+    @Override
+    public Aggregation.Result[] execute() throws IOException {
+        Aggregation.Result[] ret = null;
+        switch (readPolicy) {
+            case FORCE_LOCAL:
+                ret = getCached();
+                break;
+            case FORCE_NETWORK:
+            case BOTH:
```

Comment:
Does `getCached()` get called here if the `readPolicy` is `BOTH`?

Reply:
CachedCallback for StoreType.CACHE is processed in place where CalculationRequest is created https://github.com/Kinvey/java-library/blob/53d5c6ec09cc89d5b0b7ffd439ccee9c99868308/java-api-core/src/com/kinvey/java/store/BaseDataStore.java#L402